File: AjaxVectorObjects/Common/RedoUndo/VObjectRemovedCommand.cs
// Copyright (c) 2018 Aurigma Inc. All rights reserved.
// Licensed under the MIT License. See LICENSE file in the project root for full license information.
//
using System;

namespace Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo
{
    internal class VObjectRemovedCommand : VObjectChangedCommand
    {
        public VObjectRemovedCommand(VObject vObject, int vobjectIndex, int layerIndex) : base(vObject, vobjectIndex, layerIndex)
        {
        }

        public VObjectRemovedCommand()
            : base()
        {
        }

        public override void Execute(ICanvas canvas)
        {
            this.Data = canvas.Layers[LayerIndex].VObjects[this.VObjectIndex].Data;
            canvas.Layers[LayerIndex].VObjects.RemoveAt(this.VObjectIndex);
        }

        public override void UnExecute(ICanvas canvas)
        {
            Object b = Activator.CreateInstance(Type.GetType(this.ClassName));
            VObject obj = (VObject)(b);
            obj.Data = this.Data;
            canvas.Layers[this.LayerIndex].VObjects.Insert(this.VObjectIndex, obj);
        }
    }
}

File: AjaxVectorObjects/ClientScripts/RedoUndo/VObjectRemovedCommand.js
// Copyright (c) 2018 Aurigma Inc. All rights reserved.
// Licensed under the MIT License. See LICENSE file in the project root for full license information.
//
Type.registerNamespace("Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo");

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand = function (vObject, vObjectIndex, layerIndex) {
    /// <summary>This class represents the v-object removed command.</summary>
    /// <constructor>
    /// 	<exclude />
    /// </constructor>
    Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand.initializeBase(this, [vObject, vObjectIndex, layerIndex]);
}

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand.prototype = {
    execute: function (canvas, inGroup) {
        /// <summary>Executes this command.</summary>
        /// <param name="canvas" type="Aurigma.GraphicsMill.AjaxControls.VectorObjects.Canvas">The canvas to execute the command on.</param>
        Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand.callBaseMethod(this, 'unExecute', [canvas, inGroup]);
    },

    unExecute: function (canvas, inGroup) {
        /// <summary>Rolls this command back.</summary>
        /// <param name="canvas" type="Aurigma.GraphicsMill.AjaxControls.VectorObjects.Canvas">The canvas to roll the command back.</param>
        Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand.callBaseMethod(this, 'execute', [canvas, inGroup]);
    }
}

Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand.registerClass("Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectRemovedCommand",
	Aurigma.GraphicsMill.AjaxControls.VectorObjects.RedoUndo.VObjectAddedCommand);